### DISCUSSION TITLE
I can't make Local File System backup to work

### DISCUSSION THREAD

**Discussion by ghost** [requarks/wiki#4118] [Error / Bug Report]
Hi, I'm using [linuxserver](https://docs.linuxserver.io/images/docker-wikijs), docker version of wikijs and I wanted to set automatic data backup to local filesystem of host. I changed settings in GUI like [that](https://imgur.com/a/Cs7rc14).

Mappings of file systems look like that:

```
/mnt/ops_appdata/wikijs_config:/config
/mnt/ops_appdata/wikijs_data:/data
```

That gave me immediate error saying "EACCESS: Permission deniad mkdie /app/wiki/backup" - [screenshot](https://imgur.com/inOLMVg)

Looking at the host's filesystem, permissions and ownership looks like this:

```
ubuntu@ubuntu:/mnt/ops_appdata$ ls -al
drwxrwxrwx  2 ubuntu ubuntu  4096 Jun 21 10:30 wikijs_config
drwxrwxrwx  4 root   root    4096 May  4 22:47 wikijs_data

ubuntu@ubuntu:/mnt/ops_appdata$ la -al ./wikijs_data/
total 16
drwxrwxrwx  4 root   root   4096 May  4 22:47 .
drwxrwxrwx 17 ubuntu ubuntu 4096 Jun 16 19:05 ..
drwxrwxrwx  2 ubuntu ubuntu 4096 Jun 21 06:00 cache
drwxrwxrwx  2 ubuntu ubuntu 4096 Jun 21 06:00 uploads
ubuntu@ubuntu:/mnt/ops_appdata$ la -al ./wikijs_data/cache/
total 1088
drwxrwxrwx 2 ubuntu ubuntu   4096 Jun 21 06:00 .
drwxrwxrwx 4 root   root     4096 May  4 22:47 ..
-rw-r--r-- 1 ubuntu ubuntu   4510 Jun 21 06:02 0de024759d33c66362ddae3c2fc786a0735ea1f0.bin
-rw-r--r-- 1 ubuntu ubuntu 278487 Jun 19 00:47 13d7ed0d21fe3fb2d12bb389fa6f61f908854d1a.dat
-rw-r--r-- 1 ubuntu ubuntu   2192 Jun 21 05:41 21d5f9ba6f5cdcaa1d475c581eab4e25dd583925.bin
-rw-r--r-- 1 ubuntu ubuntu   2709 Jun 20 14:43 2cb922496fbe8e76bf7c3f28fb730fa504f0ced5.bin
-rw-r--r-- 1 ubuntu ubuntu    306 Jun 21 05:27 5cadcb3a91b6076e97b0529f72bdabbd27c539fb.bin
-rw-r--r-- 1 ubuntu ubuntu   1788 Jun 20 14:42 65e4c4b8cb5cd568d2e4978c93a6d3a804f1c4ec.bin
-rw-r--r-- 1 ubuntu ubuntu   2135 Jun 19 08:55 6c794773bc25c937634b94a1bf4ec7e0678070e6.bin
-rw-r--r-- 1 ubuntu ubuntu 486127 Jun 21 05:41 6ca1709f9f77de8bfa04d8f18aca6cedd5a79fc5.dat
-rw-r--r-- 1 ubuntu ubuntu  38478 Jun 21 05:52 852de285304ee7540f2a1293eabdd2bac13c936d.dat
-rw-r--r-- 1 ubuntu ubuntu  74902 Jun 20 06:23 9130513c2a2d90f5a2dd34baeda4f661d5ca03c3.dat
-rw-r--r-- 1 ubuntu ubuntu  44888 Jun 21 06:00 a0a6fcdba2a0e44b235b8bd4d6a3c363ec4a6a43.dat
-rw-r--r-- 1 ubuntu ubuntu 101016 Jun 20 14:34 a9d55bd8ca22ce7f4e44627ee53ba99b7768ef00.dat
-rw-r--r-- 1 ubuntu ubuntu   2188 Jun 21 05:33 b29b5d2ce62e55412776ab98f05631e0aa96597b.bin
-rw-r--r-- 1 ubuntu ubuntu  23229 Jun 21 05:52 b807ae702c66ec34e65111af2b76e78f1e01e7da.dat
-rw-r--r-- 1 ubuntu ubuntu    315 Jun 20 04:27 dc192b12fe46f33c69ee2dba4f17d53c73de440c.bin
-rw-r--r-- 1 ubuntu ubuntu   5411 Jun 20 06:23 ee46a6adb0478edb90ad9d83a3f18b8d61f0a32a.bin
-rw-r--r-- 1 ubuntu ubuntu   4041 Jun 19 00:47 fc6b93887e61d34c7a26ac1ff46b2598a7cd6312.bin
```

I tried to change ownership of /mnt/ops_appdata/wikijs_data to user that runs docker, which is: ubuntu

ubuntu@ubuntu:/mnt/ops_appdata$ sudo chown ubuntu:ubuntu wikijs_data
ubuntu@ubuntu:/mnt/ops_appdata$ ls -al
drwxrwxrwx  2 ubuntu ubuntu  4096 Jun 21 10:30 wikijs_config
drwxrwxrwx  4 ubuntu ubuntu  4096 May  4 22:47 wikijs_data

Unfortunately, that did nothing, error still persists, backup folder was not created. I tried to change permissions of wikijs_data and subfolders to 755:

ubuntu@ubuntu:/mnt/ops_appdata$ sudo chmod -R 755 wikijs_data
ubuntu@ubuntu:/mnt/ops_appdata$ ls -al wikijs_data
total 16
drwxr-xr-x  4 ubuntu ubuntu 4096 May  4 22:47 .
drwxrwxrwx 17 ubuntu ubuntu 4096 Jun 16 19:05 ..
drwxr-xr-x  2 ubuntu ubuntu 4096 Jun 21 11:02 cache
drwxr-xr-x  2 ubuntu ubuntu 4096 Jun 21 06:00 uploads

After this step my instance of wiki started to freak out - I had some issues with saving changes made to pages, so I reverted permission changes and now it seems to work finbe, but still I would like to have that backups, so please help me understand what the initial error is all about/

**Comment by NGPixel** [comment]
You need to use the path /wiki/data/content inside the docker container. This is a pre-defined volume that you can map to. Otherwise you will run into users mismatch permission issues.

**Reply by ghost** [reply]
What do you mean to use path /wiki/data/content? Should I set it in Target Configuration in Local File System settings? If so, then it is still not working - same error.

**Reply by garykuok** [reply]
Just use this path ```/wiki/data/content``` as below:
<img width="589" alt="image" src="https://user-images.githubusercontent.com/13715478/140061549-c3af4bf8-0e86-4849-87fa-31ac01fa6c44.png">

I can now backup the files to this directory. Note that do not add the last trailing slash. 
As an alternative you can try using pgdump to dump the whole database for backup: [Transing Wiki.js between servers](https://docs.requarks.io/install/transfer). However, I think this just dumps the database, but not for other files e.g., uploaded media.

FYI, I use docker-compose of wiki.js and postgre-sql on a linux VM (not linuxserver)
